pkg/sql/jobs/jobs.go

```diff
@@ -308,11 +319,26 @@ func (j *Job) Failed(ctx context.Context, err error) {
 	j.registry.unregister(*j.id)
 }
 
+type onFailer interface {
+	onFail(context.Context, *client.Txn, *Job) error
+}
+
+var _ onFailer = &Payload_Restore{}
+
+// RestoreFailHook is the func that is run when a RESTORE job has failed.
+var RestoreFailHook func(context.Context, *client.Txn, *cluster.Settings, *RestoreDetails) error
+
+func (r *Payload_Restore) onFail(ctx context.Context, txn *client.Txn, job *Job) error {
+	// This is set in the CCL code if RESTOREs are enabled, so no need to check
+	// for nil.
```

Comment:
Heh, looks like this isn't true when running the tests for the jobs package. 🤦‍♂️